specs/DesignDiscussions/InputValidation.md

```diff
@@ -0,0 +1,503 @@
+
+![img](assets/InputValidation/FoodOrder.gif)
+
+#### Only validate visible fields
+
+One option would be to only validate fields that are currently visible. This introduces a number of concerns. The first is that currently we send all inputs (visible or not) on a submit action. This means that given the current behavior we would be sending potentially non-validated inputs back to the card author. With client side input validation, the code these are ultimately sent to may expect that they've already been validated. Not validating them would break that expectation.
+
+If we decide not to even *send* the hidden (and therefore not-validated) inputs, this would be a breaking change as today we send *all* inputs on submit.
+
+Finally, as the visibility of an input is ultimately determined by the user, we don't want hiding the input to bypass validation. In the below scenario, closing the personalization card should not allow you to use special characters (nor should it imply you no longer want it personalized).
+
+![img](assets/InputValidation/InternalShowCardValidation.gif)
+
+#### Making hidden element visible on submit
+In the above card, showing the collapsed show card at submit would allow us to show the user the errors that are blocking submit. While this may work well in some simple ShowCard cases such as the this one, there are a number of pitfalls. A card that makes use of toggle visibility may have non-trivial logic governing what is shown and what is not shown at any time. Consider the following card built using toggle visibility:
+
+![img](assets/InputValidation/ToggleWizard.gif)
+
+If there was a validation error on the first "page", bringing the hidden element back into focus on submit would result in one of the following, depending on whether we brought back the entire hidden element or just the invalid input. In either case, this is not the expected behavior of this card author, and the subsequent behavior if the toggle buttons are pressed again would not be well defined:
+
+Bring back non-visible card element            |  Bring back single input
+:-------------------------:|:-------------------------:
+![img](assets/InputValidation/BrokenToggleWizard.PNG)  |  ![img](assets/InputValidation/BrokenToggleWizard2.PNG)
+
+> Note on this card: This "Wizard" scenario is implemented here via toggle visibility. We should consider whether we want to support a first class wizard experience where a new element type could list a series of subcards to display in order. Whether or not we ever introduce such a concept, the point remains that toggle logic may be complex, and we shouldn't be arbitrarily toggling things visible outside the scope of what the card author has defined in their toggle buttons.
+
+#### Current Card/Container
```

Comment:
This is effectively the approach you land on at the end, so maybe combine, reorganize, or mention that here. Also, this section brings up concerns with this approach that aren't addressed in the conclusion, so we should probably add something around how we're addressing them and/or why they're less pressing than other considerations.